### PR TITLE
Report issue#88

### PR DESCRIPTION
**Pull request by KonstantinGasser**
This is a first proposal for the `report` command.
Supported flags:
- `--billable`
- `--project`
- `--start` / `--end`
- `--format` (json/csv) csv is not yet implemented...
- `--out`

The command is designed the way it is discussed in #88.

In the `cli/report.go` I commented TODOs. Here I would be happy to get some input as I don't like the way I have implemented the checks if a `Flags.StringVarP` has its default value...

### CODE REVIEW

**Comment by KonstantinGasser** [comment]
@dominikbraun while using the report feature my self I found a bug causing a panic if `--start` or `--end` where given.. I have fixed that bug and appended it to this PR. I also fixed another bug where the `--end` time would not be inclusive.

**Comment by dominikbraun** [comment]
Thanks! Is this ready to review already?

**Comment by KonstantinGasser** [comment]
Yes it is ready to be reviewed - all of the above mentioned flags are supported (only serialisation to csv is not implemented yet)

**Comment by KonstantinGasser** [comment]
@dominikbraun I noticed a merge conflict for this PR after the latest release - I fixed that one PR is ok again.

**Review by dominikbraun — COMMENTED** [review]
Looks pretty good so far, I'm going to test it locally today.

File: core/record.go [hunk 1]
```diff
@@ -296,6 +297,36 @@ func (t *Timetrace) loadOldestRecord(date time.Time) (*Record, error) {
...
+		for _, f := range filter {
+			if !f(record) {
+				useRecord = false
+				break
+			}
+		}
+		// check if eighter filter negates useRecord
+		if !useRecord {
+			continue
+		}
```
[the remark this answers is not in the document]
Comment (KonstantinGasser) [reply core/record.go:324]: That's why there are Reviews! :D Didn't thought of that - I will implement that.

**Comment by KonstantinGasser** [comment]
@dominikbraun I have Uni until 6pm I will implement your changes afterwards and update the PR 👍

File: core/formatter.go [hunk 2]
```diff
@@ -47,6 +47,17 @@ func (f *Formatter) TimeString(input time.Time) string {
...
 
+// PrettyDateString returns a nice representation of a given time
+// example: Mon 31. May 2021
+func (f *Formatter) PrettyDateString(input time.Time) string {
+	year := input.Local().Year()
+	day := input.Local().Day()
+	weekday := input.Local().Weekday()
+	month := input.Local().Month()
+
+	return fmt.Sprintf("%v %d. %v %d", weekday, day, month, year)
```
Comment (dominikbraun) [inline core/formatter.go:58]: You can replace this with

```suggestion
	return input.Format("Monday, 02. January 2006")
```

**Comment by KonstantinGasser** [comment]
I have incorporated your suggestions and updated the PR

**Review by dominikbraun — APPROVED** [review]
Thank alot for implementing this feature!

It will be included in one of the next releases and officially published in its very own release.

**Comment by KonstantinGasser** [comment]
Cool, great to hear!

**Review by dominikbraun — APPROVED** [review]
Will be merged and included in timetrace v0.11.0 as an hidden feature.

**Pull request merged**